src/hooks.ts

```diff
@@ -51,66 +56,47 @@ interface UseFeature {
 }
 
 /**
- * A React Hook that retrieves the status of a feature flag and its variables, optionally
- * auto updating those values based on underlying user or datafile changes.
- *
- * Note: The react client can become ready AFTER the timeout period.
- *       ClientReady and DidTimeout provide signals to handle this scenario.
+ * A function which waits for the optimizely client instance passed to become
+ * ready and then sets up initial state and (optionally) autoUpdate listeners
+ * for the hook type specified.
  */
-export const useFeature: UseFeature = (featureKey, options = {}, overrides = {}) => {
-  const { isServerSide, optimizely, timeout } = useContext(OptimizelyContext);
-  if (!optimizely) {
-    throw new Error('optimizely prop must be supplied via a parent <OptimizelyProvider>');
-  }
-  const finalReadyTimeout: number | undefined = options.timeout !== undefined ? options.timeout : timeout;
-
-  // Helper function to return the current values for isEnabled and variables.
-  const getCurrentValues = useCallback(
-    () => ({
-      isEnabled: optimizely.isFeatureEnabled(featureKey, overrides.overrideUserId, overrides.overrideAttributes),
-      variables: optimizely.getFeatureVariables(featureKey, overrides.overrideUserId, overrides.overrideAttributes),
-    }),
-    [featureKey, overrides]
-  );
-
-  // Set the initial state immediately serverSide
-  const [data, setData] = useState<UseFeatureState>(() => {
-    if (isServerSide) {
-      return getCurrentValues();
-    }
-    return { isEnabled: false, variables: {} };
-  });
-
-  const [clientReady, setClientReady] = useState(isServerSide ? true : false);
-  const [didTimeout, setDidTimeout] = useState(false);
-
-  useEffect(() => {
+const initializeWhenClientReadyFn = (
+  type: HookType,
+  name: string,
+  optimizely: ReactSDKClient,
+  options: UseFeatureOptions,
+  logger: LoggerFacade,
```

Comment:
Why do we need to pass a logger in? So far the convention seems to be that each module has its own logger. If the goal is to distinguish different callers by the log message, logging the `type` could work for that.